Geta.EPi.Extensions/UrlExtensions.cs

```diff
@@ -75,5 +75,45 @@ private static Uri GetBaseUri()
                 ? SiteDefinition.Current.SiteUrl
                 : new Uri(HttpContext.Current.Request.Url.GetLeftPart(UriPartial.Authority));
         }
+
+        /// <summary>
+        ///     Gets the Url type of provided Url
+        /// </summary>
+        /// <param name="url"></param>
+        /// <returns>The type of Url (Internal, External). Returns Unknown if resolution fails.</returns>
+        public static UrlType GetUrlType(this Url url)
+        {
+            if (url.IsAbsoluteUri)
```

Comment:
Isn't it so that even link specified in EPiServer as external gets overwritten by EPiServer and converted to internal - resulting in relative url? Need to test.. But in general - might be an issue here, yes.